Sources/OpenAPIRuntime/Multipart/MultipartFramesToRawPartsSequence.swift

```diff
@@ -342,8 +342,17 @@ extension MultipartFramesToRawPartsSequence {
             switch stateMachine.nextFromPartSequence() {
             case .returnNil: return nil
             case .fetchFrame:
+                let frame: Upstream.AsyncIterator.Element?
                 var upstream = upstream
-                let frame = try await upstream.next()
+                if #available(macOS 15, iOS 18.0, tvOS 18.0, watchOS 11.0, macCatalyst 18.0, visionOS 2.0, *) {
+                    #if compiler(>=6.0)
+                    frame = try await upstream.next(isolation: self)
+                    #else
+                    frame = try await upstream.next()
+                    #endif
+                } else {
+                    frame = try await upstream.next()
+                }
```

Comment:
While we're here, would you mind creating an extension on `AsyncIteratorProtocol` that adds a utility method for this dance, and call it from these two places you're updating in the PR? It'd take the actor as a parameter.